feat(B-143): submit user feedback

annaschleich opened the pull request:
# Descriptive Title
### Current Situation
Closes: #143 
### Proposed Solution
- adds new database table for user feedbacks
- provides endpoints to retrieve feedbacks and submit a feedback
#### Implications
- this PR is provides functionality for #142 and #144
### Testing
- local testing with Postman
### Reviewer Notes
- nothing special

rautenrieb reviewed (changes requested):
See new [naming conventions](https://www.notion.so/Naming-Conventions-20a5f448405780168d91db483c284345?source=copy_link). Let's rename this to `app review`.

rautenrieb commented on backend/app/routers/app_review_route.py:
```diff
@@ -0,0 +1,47 @@
+)
+from app.models.user_profile import UserProfile
+
+router = APIRouter(prefix='/feedback', tags=['User Feedback'])
```
Let's rename this to /app-review and the tag to 'App Review'

rautenrieb commented on backend/app/routers/app_review_route.py:
```diff
@@ -0,0 +1,47 @@
+@router.get('/', response_model=list[AppReviewRead])
+def get_app_reviews(session: Annotated[Session, Depends(get_session)]) -> list[AppReview]:
+    """
+    Retrieve all user feedbacks.
```
> Retrieve all user app reviews.

rautenrieb commented on backend/app/routers/app_review_route.py:
```diff
@@ -0,0 +1,47 @@
+    app_review: AppReviewCreate, session: Annotated[Session, Depends(get_session)]
+) -> AppReviewResponse:
+    """
+    Create a new user feedback.
```
> Create a new user app review.

rautenrieb reviewed (changes requested):
Small naming changes. Otherwise functionality looks good!

annaschleich commented:
Renamed enpoint to /review and added session_id as an optional field for a review. See notion discussion: https://www.notion.so/Get-Recent-Reviews-2015f4484057803e824ac89fe9ced585?d=20c5f4484057801aa2db001cd8562af7&source=copy_link

rautenrieb commented on backend/app/models/review.py:
```diff
@@ -0,0 +1,38 @@
+class Review(SQLModel, table=True):  # `table=True` makes it a database table
+    id: UUID = Field(default_factory=uuid4, primary_key=True)
+    user_id: UUID = Field(foreign_key='userprofile.id', nullable=False)  # FK to UserProfile
+    session_id: UUID = Field(
```
Make this optional. Getting a type warning in the dummy data.

Something like:
```Python
session_id: UUID | None = Field(
        default=None, foreign_key='trainingsession.id', nullable=True
    )  # FK to TrainingSession, optional
```

rautenrieb reviewed (approved):
Almost done. Just see my comment about make the session id optional in the model.

rautenrieb reviewed (changes requested):
Sorry, meant to request changes...

rautenrieb reviewed (approved):
Thanks Anna. Ready to merge now! 🚀
